torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1919,6 +2001,18 @@ def relation_with_digit(expr, op, digit):
                     assert op == "=="
                     results[expr]["eq"] = digit
 
+            # retrieve dynamic shapes
+            name_to_dim = {}
+            for dim in pytree.tree_flatten(
```

Comment:
Isn't there a different way to collect all dims? Like, have you checked the contents of `source_name_to_debug_name`?

Reply:
I think the main issue here is collecting root dims which aren't attached to a size